### DISCUSSION TITLE
TDS Initialization sequence

### DISCUSSION THREAD

**Discussion by jinningwang** [CURENT/andes#179] [Internal Design] [answered]
Hi Hantao,

I have two questions about model initialization.
1) ``ConstService`` is ahead of ``VarService``, right?
2) Let say we have a ``A = ConstService(...)`` and  a ``B = VarService(...)``. If ``A`` calls ``B`` ahead of the definition of ``B``, then ``B`` will appear as 0 in ``A``? 

Regards,
Jinning

**Comment by cuihantao** [comment, marked as answer]
ConstService and VarService are initialized at the same time, based on their sequence of definition.

 

In this case, If A uses B, B will appear as 0 in A. It is the same case even if both A and B are ConstService.

**Reply by jinningwang** [reply]
Thanks for your reply!

Regards,
Jinning
